astropy/tests/helper.py

```diff
@@ -470,3 +471,33 @@ def assert_quantity_allclose(actual, desired, rtol=1.e-7, atol=None,
     from astropy.units.quantity import _unquantify_allclose_arguments
     np.testing.assert_allclose(*_unquantify_allclose_arguments(
         actual, desired, rtol, atol), **kwargs)
+
+
+def assert_type_is_type(a, b):
+    """Assert that two objects have exactly the same type.
+
+    Parameters
+    ------------
+    a, b : object or type
+        The two objects to compare. If either is a `type`,
+         then compare directly to that type instead.
+
+    Raises
+    -------
+    AssertionError
+        If the type of ``a`` is not the type of ``b``.
+
+    Examples
+    ----------
+    The following pass:
+
+        >>> assert_type_is_type(int, int)
+        >>> assert_type_is_type(1, int)
+        >>> assert_type_is_type(int, 1)
+        >>> assert_type_is_type(1, 2)
+    """
+
+    t_a = a if (t_a := type(a)) is type else t_a
```

Comment:
This is not quite good enough:
```
In [9]: type (Time)
Out[9]: abc.ABCMeta
```
I think you actually want `isinstance(type(a), type)` :smiley_cat: